Install a new event loop explicitly

jeremycline opened the pull request:
Python introduced some breaking changes[0] which cause it to no longer implicitly create an event loop. Create a new one and install it explicitly so Twisted doesn't collapse at import.

Note that Python plans to remove more APIs so this will certainly break in the future.

[0] https://docs.python.org/3/whatsnew/3.14.html#id10

Fixes: #531

Take 2 with (hopefully) a "fix" for pytest-twisted

jeremycline committed:
Install a new event loop explicitly

Python introduced some breaking changes[0] which cause it to no longer
implicitly create an event loop. Create a new one and install it
explicitly so Twisted doesn't collapse at import.

Note that Python plans to remove more APIs so this will certainly break
in the future.

[0] https://docs.python.org/3/whatsnew/3.14.html#id10

Fixes: #531

Signed-off-by: Jeremy Cline <jeremycline@linux.microsoft.com>

abompard reviewed (approved):
LGTM! You only forgot to add 3.14 to the classifiers in pyproject.toml but that's really minor. Thanks a lot!

jeremycline commented:
> LGTM! You only forgot to add 3.14 to the classifiers in pyproject.toml but that's really minor. Thanks a lot!

Thanks for the review! I've added the classifier and rebased. Do you want to do a release with this, or should be backport just this patch for a F43 build?

abompard commented:
Nah I'll do a release, thanks.
